### ISSUE TITLE
Player placing objects system

### ISSUE THREAD

**Issue by KieranClarkee** [PlaceholderGames/2016-Team2Thirds#4]
This issue will be for the system that we need to put in place so that the third person player will be able to place different objects, or traps into the game.

**Comment by KieranClarkee** [comment]
I have managed to create a system that allows a third person player place a floor piece and a wall piece. I have done this by first creating a building trace function that gets the actors location, breaks the vector then create another one with the distance away from the player added to it. 

Within the event graph I have added the input information that details what happens when each key is pressed, which includes choosing either a floor or a floor, rotating the object and then toggling the placement.

Also within the event graph there is a ToggleBuildMode that is used to determine if the player is still in the build mode or not. This event connects to the Enum of the two objects which allows their blueprints to be connected to. This event is then branched so if false, it sets "In building mode and placed" to false which allows the player to spawn the actor. From this another branch is created and the "Placed" boolean is connected as the condition. This means that if the placed boolean is true then it sets the "in build mode" to false which stops the actor from continuing to drag around the object after it has been placed.
